zproject/wsgi.py

```diff
@@ -33,6 +36,37 @@
     # setting points here.
 
     application = get_wsgi_application()
+
+    # We force loading of the main parts of the application now, by
+    # handing it a fake request, rather than have to pay that price
+    # during the first request served by this process.  Hitting the
+    # /health endpoint will not only load Django and all of the views
+    # (by loading the URL dispatcher) but will also force open any
+    # lazy-loaded service connections.
+    #
+    # The return value (and thus response status) of this healthcheck
+    # request is ignored, so we do return the application handler even
+    # if connections are not fully available yet.  This at least
+    # allows application logging to handle any such errors, instead of
+    # arcane errors from uwsgi not being able to load its handler
+    # function.
+    def ignored_start_response(
+        status: str, headers: list[tuple[str, str]], exc_info: Any = None, /
+    ) -> Callable[[bytes], object]:
+        return lambda x: None
+
+    application(
+        {
+            "REQUEST_METHOD": "GET",
+            "SERVER_NAME": "127.0.0.1",
+            "SERVER_PORT": "80",
```

Comment:
Is 80 the right port to simulate? I guess 80/https seems inconsistent to me.

Reply:
Not as inconsistent as you might think -- see 0935d388f05337e73898d932ddb75b458cf4747e.  Specifically, uwsgi has an undocumented feature where it [sets `wsgi.url_scheme` based on `X-Forwarded-Proto`](https://github.com/unbit/uwsgi/blob/73efb013e94636d8ff7d0e375da0d6d0595dbdac/core/protocol.c#L558-L561), meaning that this is the combination of port and scheme that Django would see if it was running on port 80 behind a TLS-terminating proxy.

However, for clarity, I'll adjust to port 443.